### PR TITLE
feat(skills): add Atheris fuzz harness with CI workflow integration

### LINKED ISSUE #1021: feat(powerpoint): add thin Atheris wrapper for Scorecard Fuzzing compliance
## Summary

Add a thin Atheris wrapper using the polyglot pattern to satisfy the OSSF Scorecard Fuzzing check. The Scorecard check only recognizes `import atheris` for Python fuzzing — Hypothesis alone scores 0/10 despite providing genuine property-based testing value. This issue creates a bridge between the Hypothesis test infrastructure (Phase 1) and Scorecard's detection requirements.

## Context

This is **Phase 3** of the Python Security Testing & Fuzzing Initiative.

### OSSF Scorecard Fuzzing Detection Pipeline

Scorecard uses a three-phase detection pipeline with binary scoring (any fuzz detection = 10/10):
1. `.clusterfuzzlite/Dockerfile` with non-comment content
2. `google/oss-fuzz` project registration
3. Language-specific import patterns — for Python: only `import atheris`

Hypothesis is recognized for Haskell (QuickCheck), JavaScript (fast-check), C# (FsCheck), Erlang (eqc), and others — but **not for Python**. This is a known gap in Scorecard's detection.

### Prominent Language Filter

Scorecard only scans "prominent" languages (LoC >= 25% of average). Python may or may not meet the threshold in this predominantly Markdown/PowerShell/YAML repository.

### Polyglot Pattern

Create a single Python file that works both as a pytest property test AND as an Atheris fuzz target:

```python
"""Polyglot fuzz harness: runs as Hypothesis property test in CI, Atheris fuzz target for Scorecard."""
import sys

try:
    import atheris
    FUZZING = True
except ImportError:
    FUZZING = False

from hypothesis import given, settings
import hypothesis.strategies as st

yaml_values = st.recursive(
    st.none() | st.booleans() | st.integers() | st.floats(allow_nan=False) | st.text(),
    lambda children: st.lists(children) | st.dictionaries(st.text(), children),
    max_leaves=50,
)

@given(data=yaml_values)
@settings(max_examples=500)
def test_parse_yaml_input(data):
    """Polyglot: runs as Hypothesis property test or Atheris fuzz target."""
    # Property test logic here — reuses strategies from Phase 1
    ...

if FUZZING:
    atheris.Setup(sys.argv, atheris.instrument_func(test_parse_yaml_input))
    atheris.Fuzz()
```

### File Location

`.github/skills/experimental/powerpoint/tests/fuzz_harness.py` — placed in the test directory to be discovered by Scorecard's file scanning.

### Dependency

`atheris` should be an optional dependency (not required for CI):

```toml
[dependency-groups]
fuzz = [
    "atheris>=2.3.0",
]
```

## Dependencies

- **Depends on**: Hypothesis property tests (Phase 1) — reuses strategies and property definitions
- **Related**: Contribute Hypothesis detection to ossf/scorecard (upstream alternative)

### task-researcher
- Confirm Scorecard's file scanning scope for this repository (does it scan `.github/skills/` subtrees?)
- Verify the prominent language filter threshold for Python in this repo
- Test whether `import atheris` in a try/except block satisfies Scorecard's regex detection
- Check Atheris compatibility with Python >=3.11 and the project's test infrastructure

### task-planner
- Design the polyglot harness to maximize reuse of Phase 1 Hypothesis strategies
- Determine file placement that satisfies Scorecard scanning
- Plan Atheris as an optional dependency (not required for CI)
- Evaluate if Phase 1 completion is a hard prerequisite or if a minimal harness can be created independently

### task-implementor
- Create the polyglot fuzz harness file
- Add `atheris>=2.3.0` as an optional dependency group
- Verify the harness works as a standalone pytest test (Hypothesis mode)
- Verify Scorecard detection by checking the `import atheris` pattern match
- Document the polyglot pattern and its purpose

## Acceptance Criteria

- [ ] A Python file containing `import atheris` exists in a location scanned by Scorecard
- [ ] The file works as a valid pytest test when Atheris is not installed (Hypothesis-only mode)
- [ ] The file works as an Atheris fuzz target when Atheris is installed
- [ ] Atheris is an optional dependency, not required for standard CI
- [ ] The polyglot harness reuses Hypothesis strategies from Phase 1 property tests
- [ ] OSSF Scorecard Fuzzing check detects the Atheris import (verified via local Scorecard run or manual regex check)
- [ ] The harness is documented with its dual-mode purpose

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
## Description

Added a polyglot Atheris fuzz harness convention to the powerpoint skill, established it as a required pattern for all Python skills with tests, documented the convention across security and contributing guides, and integrated coverage-guided fuzzing into the PR validation pipeline. The fuzz harness immediately surfaced two crash bugs in *pptx_colors.py* that were fixed in the same changeset. A shell script bug in *generate.sh* was also discovered and fixed during the PR workflow.

### Fuzz Harness Implementation

&gt; OSSF Scorecard Fuzzing detection requires an `import atheris` statement reachable by its Phase 3 regex scanner. The polyglot harness pattern satisfies this while keeping CI-friendly pytest tests as the default execution mode.

- Added **fuzz_harness.py** (222 lines) to the powerpoint skill with 4 fuzz targets covering `resolve_color`, `hex_brightness`, `max_severity`, and `_has_formatting_variation`
  - Dual-mode design: pytest tests (21 tests across 4 test classes) run everywhere; Atheris fuzzing activates only when the package is available
  - `fuzz_dispatch` router uses first byte modulo target count to distribute fuzzer input across targets
- Added `fuzz = [&quot;atheris&gt;=3.0&quot;]` dependency group to *pyproject.toml*, intentionally separate from `dev` since Atheris only ships manylinux wheels
- Added `python_files = [&quot;test_*.py&quot;, &quot;fuzz_harness.py&quot;]` to pytest discovery configuration

### CI Workflow Integration

- Added **fuzz-tests.yml** reusable workflow for coverage-guided Atheris fuzzing in PR validation
  - Discovers Python projects via existing `discover-python-projects` job and gates on `fuzz_harness.py` presence
  - Runs configurable fuzz iterations (default 10,000) with `uv sync --locked --dev --group fuzz`
  - Produces GitHub Actions step summary with run statistics (iterations, edges, features, corpus size, crashes)
  - Uploads crash artifacts and JSON results per fuzz project for traceability
  - Supports `soft-fail` and `changed-files-only` inputs for flexible pipeline configuration
  - Follows established workflow conventions (pwsh shell, three-dot diff, artifact name sanitization)
- Integrated **fuzz-tests** job into **pr-validation.yml** with matrix strategy across discovered Python projects

### Bug Fixes

- Fixed **short hex string crashes** in *pptx_colors.py* discovered by the fuzz harness
  - `resolve_color` now returns a safe default RGB when hex strings are shorter than 6 characters
  - `hex_brightness` now returns `0` for short hex strings instead of silently producing incorrect values
- Fixed **empty pathspec bug** in *generate.sh* where `printf '%s\n'` on an empty array produced a blank newline, causing `git diff` to fail with `fatal: empty string is not a valid pathspec`
  - Wrapped in an array length guard: `if [[ ${#specs[@]} -gt 0 ]]`

### Validation &amp; Documentation

- Added **error-level validation** in *Validate-SkillStructure.ps1* checking that Python skills with a `tests/` directory include `tests/fuzz_harness.py`
- Updated *.github/copilot-instructions.md* with 3 convention entries for the fuzz harness requirement
- Added *docs/security/fuzzing.md* covering detection strategy, running instructions, and platform compatibility
- Updated *docs/contributing/skills.md* with a fuzz harness section, directory tree update, and 2 submission checklist items
- Updated *docs/security/README.md* and *scripts/linting/README.md* with index entries for the new content

## Related Issue(s)

Closes #1021

## Type of Change

Select all that apply:

**Code &amp; Documentation:**

* [ ] Bug fix (non-breaking change fixing an issue)
* [x] New feature (non-breaking change adding functionality)
* [ ] Breaking change (fix or feature causing existing functionality to change)
* [x] Documentation update

**Infrastructure &amp; Configuration:**

* [x] GitHub Actions workflow
* [ ] Linting configuration (markdown, PowerShell, etc.)
* [ ] Security configuration
* [ ] DevContainer configuration
* [ ] Dependency update

**AI Artifacts:**

* [ ] Reviewed contribution with `prompt-builder` agent and addressed all feedback
* [x] Copilot instructions (`.github/instructions/*.instructions.md`)
* [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
* [ ] Copilot agent (`.github/agents/*.agent.md`)
* [x] Copilot skill (`.github/skills/*/SKILL.md`)

**Other:**

* [x] Script/automation (`.ps1`, `.sh`, `.py`)
* [ ] Other (please describe):

## Testing

- Ran `npm run validate:skills` — fuzz harness validation passes for the powerpoint skill
- Ran `npm run lint:ps` — PowerShell analysis passes for Validate-SkillStructure.ps1 changes
- Ran `npm run lint:md` — markdown linting passes for all modified documentation files
- Ran `npm run spell-check` — spelling validation passes
- Ran `npm run lint:frontmatter` — frontmatter validation passes
- Ran `npm run lint:md-links` — markdown link checking passes
- Ran `actionlint` — workflow validation passes for both fuzz-tests.yml and pr-validation.yml
- Ran `yaml-lint` — YAML validation passes for fuzz-tests.yml
- Fuzz harness pytest mode executed successfully with all 21 tests passing
- Local fuzz run completed 10,000 iterations with 0 crashes (~66s wall-clock)

### Required Checks

* [x] Documentation is updated (if applicable)
* [x] Files follow existing naming conventions
* [x] Changes are backwards compatible (if applicable)
* [ ] Tests added for new functionality (if applicable)

### AI Artifact Contributions

* [ ] Used `/prompt-analyze` to review contribution
* [ ] Addressed all feedback from `prompt-builder` review
* [ ] Verified contribution follows common standards and type-specific requirements

### Required Automated Checks

The following validation commands must pass before merging:

* [x] Markdown linting: `npm run lint:md`
* [x] Spell checking: `npm run spell-check`
* [x] Frontmatter validation: `npm run lint:frontmatter`
* [x] Skill structure validation: `npm run validate:skills`
* [x] Link validation: `npm run lint:md-links`
* [x] PowerShell analysis: `npm run lint:ps`

## Security Considerations

No sensitive data, credentials, or privilege escalation paths introduced. The Atheris dependency is isolated in a `fuzz` group that is not installed by default. The `pptx_colors.py` fixes close input-validation gaps that could have caused unexpected behavior with malformed color strings. The fuzz-tests workflow runs with `contents: read` permissions only.

## Additional Notes

- The `fuzz` dependency group is intentionally separate from `dev` because Atheris only ships manylinux wheels (no macOS). This prevents `uv sync` failures on non-Linux platforms.
- The *uv.lock* update includes a `cairosvg` wheel addition that appears to be a pre-existing lock drift correction.
- The fuzz-tests workflow aligns with established conventions from python-lint.yml and pip-audit.yml (pwsh shell defaults, three-dot diff for changed-files detection, artifact name sanitization, GITHUB_ENV gating).

### CODE REVIEW

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+try:
+    import atheris
+
+    FUZZING = True
+except ImportError:
+    FUZZING = False
+
+from extract_content import _has_formatting_variation
+from pptx_colors import hex_brightness, resolve_color
+from validate_deck import max_severity
```
Comment (katriendg) [inline .github/skills/experimental/powerpoint/tests/fuzz_harness.py:23]: ⚠️ **`atheris.instrument_imports()` is called after target module imports — coverage guidance is inactive**

`atheris.instrument_imports()` runs inside the `if __name__ == "__main__"` block (line 228), which executes after all module-level imports have already been resolved and cached in `sys.modules`. Atheris hooks Python's import loader to inject libFuzzer coverage callbacks into bytecode at load time — modules already imported are silently skipped.

In practice the fuzzer runs, but no branch coverage from `pptx_colors`, `extract_content`, or `validate_deck` is fed back to libFuzzer, so it degrades to undirected random mutation and is unlikely to rediscover the short-hex bugs or reach deep paths.

The standard fix is to call `instrument_imports()` as a context manager **before** the SUT imports:

```python
try:
    import atheris
    FUZZING = True
except ImportError:
    FUZZING = False

if FUZZING:
    with atheris.instrument_imports():
        from extract_content import _has_formatting_variation
        from pptx_colors import hex_brightness, resolve_color
        from validate_deck import max_severity
else:
    from extract_content import _has_formatting_variation
    from pptx_colors import hex_brightness, resolve_color
    from validate_deck import max_severity

# ...

if __name__ == "__main__" and FUZZING:
    # instrument_imports() removed from here
    atheris.Setup(sys.argv, fuzz_dispatch)
    atheris.Fuzz()
```

The duplicate import block is verbose but required for the polyglot pattern to work correctly under both pytest and Atheris modes.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    fuzz_has_formatting_variation,
+]
+
+
+def fuzz_dispatch(data):
+    """Route Atheris input to one of the registered fuzz targets."""
+    if len(data) < 2:
+        return
+    idx = data[0] % len(FUZZ_TARGETS)
+    FUZZ_TARGETS[idx](data[1:])
```
Comment (katriendg) [inline .github/skills/experimental/powerpoint/tests/fuzz_harness.py:121]: 💡 **Fuzz target functions reference `atheris` without a guard — semantic trap for future contributors**

All four `fuzz_*` functions call `atheris.FuzzedDataProvider(data)` directly. When `FUZZING = False` (no Atheris installed), `atheris` is not bound in the module namespace, so invoking any `fuzz_*` function raises `NameError`. This is safe today because pytest only discovers `test_*`/`Test*`-prefixed names. However, a future contributor who calls a fuzz helper from a regular test would hit a confusing `NameError` with no hint about why.

Consider annotating the fuzz function block to make the intent explicit:

```python
# Fuzz targets below are only callable when FUZZING = True (Atheris installed).
# They are not discovered by pytest due to the fuzz_* prefix.
```

Or add a `# pragma: no cover` comment on each function to exclude them from coverage reports, since they are unreachable in the CI path.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py
Comment (katriendg) [inline .github/skills/experimental/powerpoint/tests/fuzz_harness.py:1]: 💡 **Consider adding a corpus seed directory**

LibFuzzer starts from random bytes when no corpus exists, which means early runs often fail to reach interesting branches. A small set of hand-crafted edge-case inputs dramatically improves initial coverage and shortens the time to find new bugs.

Creating a `tests/corpus/` directory with a few seed files is near-zero effort:

```bash
mkdir -p tests/corpus
printf '\x00#'         > tests/corpus/short-hex       # triggers short-hex guard
printf '\x01#GGGGGG'  > tests/corpus/invalid-hex      # non-hex chars
printf '\x02@accent1' > tests/corpus/theme-ref        # theme lookup path
printf '\x03'         > tests/corpus/empty            # empty input guard
```

Then invoke with: `uv run python tests/fuzz_harness.py tests/corpus/`

This is optional for a Scorecard-compliance harness but is the standard practice for any fuzzer expected to run beyond CI.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        ]
+        slides.append({"issues": issues})
+    num_deck_issues = fdp.ConsumeIntInRange(0, 3)
+    deck_issues = [
+        {"severity": severities[fdp.ConsumeIntInRange(0, len(severities) - 1)]}
+        for _ in range(num_deck_issues)
+    ]
+    results = {"slides": slides, "deck_issues": deck_issues}
+    max_severity(results)
+
```
Comment (katriendg) [inline .github/skills/experimental/powerpoint/tests/fuzz_harness.py:83]: 💡 **`fuzz_max_severity` does not probe missing-key resilience**

The generated `results` dict always contains both `slides` and `deck_issues` keys. If `max_severity` uses direct subscript access (`results["slides"]`) a dict missing that key would crash — this fuzz target would not discover it because the router always constructs a fully-formed dict.

Worth adding variants that omit one or both keys to verify `max_severity` is robust to partial input:

```python
# Inside fuzz_max_severity, after the main results dict:
if fdp.ConsumeBool():
    max_severity({"slides": slides})            # missing deck_issues
if fdp.ConsumeBool():
    max_severity({"deck_issues": deck_issues})  # missing slides
max_severity({})                                # empty dict
```

File: docs/security/fuzzing.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+OSSF Scorecard evaluates fuzzing through a three-phase detection pipeline:
+
+| Phase | Detection Method                    | HVE Core Strategy                              |
+|-------|-------------------------------------|------------------------------------------------|
+| 1     | OSSFuzz integration                 | Not applicable (HVE Core is not a library)     |
+| 2     | ClusterFuzzLite YAML                | Not applicable (no `.clusterfuzzlite/` config) |
+| 3     | Source file regex: `import atheris` | Active: `fuzz_harness.py` in each Python skill |
+
+Phase 3 scans all `.py` files in the repository for `import atheris`. A single match gives the repository a 10/10 Fuzzing score.
```
Comment (katriendg) [inline docs/security/fuzzing.md]: 💡 **Minor precision nit on Scorecard scoring language**

The current text reads: *"A single match gives the repository a 10/10 Fuzzing score."*

This could be read as saying the overall Scorecard score becomes 10/10. The Fuzzing check is one weighted sub-check among ~20; passing it contributes its weight to the aggregate. Suggested rewording:

> Phase 3 scans all `.py` files in the repository for `import atheris`. A single match satisfies the OSSF Scorecard Fuzzing check, contributing its full weight to the repository's overall Scorecard score.

File: .github/skills/experimental/powerpoint/pyproject.toml [hunk 5]
```diff
@@ -17,10 +17,14 @@ dev = [
     "pytest-cov>=7.0",
     "ruff>=0.15",
 ]
+fuzz = [
+    "atheris>=3.0",
+]
```
Comment (katriendg) [inline .github/skills/experimental/powerpoint/pyproject.toml:24]: 💡 **Consider a brief inline comment on the macOS/wheel constraint**

The architectural reason for a separate `fuzz` group (Atheris ships manylinux-only wheels) is documented in the PR description and `docs/security/fuzzing.md`, but a contributor reading `pyproject.toml` in isolation won't know why `atheris` is isolated from `dev`. A one-line comment makes the intent self-evident:

```toml
# Kept separate from dev — Atheris only ships manylinux wheels (no macOS/Windows).
fuzz = [
    "atheris>=3.0",
]
```

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+# Fuzz targets — pure functions exercised by both modes
+# ---------------------------------------------------------------------------
+
+
+def fuzz_resolve_color(data):
+    """Fuzz resolve_color with str and dict inputs."""
+    fdp = atheris.FuzzedDataProvider(data)
+    hex_str = "#" + fdp.ConsumeUnicodeNoSurrogates(6)
+    try:
+        resolve_color(hex_str)
```
Comment (auyidi1) [inline .github/skills/experimental/powerpoint/tests/fuzz_harness.py:35]: In `fuzz_resolve_color`, every call is wrapped in `except (ValueError, IndexError): pass`. If the SUT starts raising a different exception type (e.g., `TypeError`, `KeyError`), the harness would surface it as a crash — which is actually correct fuzzer behavior. However, the comment "expected parse failures" implies these are the only expected failures. If `resolve_color` legitimately raises `TypeError` on some inputs, the harness would report false positives.

File: scripts/linting/Validate-SkillStructure.ps1 [hunk 7]
```diff
@@ -168,6 +168,14 @@ function Test-PythonSkillConfig {
         $warnings.Add("pyproject.toml does not list ruff in dev dependencies in '$RelativePath'")
     }
 
+    # Fuzz harness convention check
+    if ($HasTestsDir) {
+        $fuzzHarnessPath = Join-Path (Split-Path $PyprojectPath -Parent) 'tests' 'fuzz_harness.py'
```
Comment (auyidi1) [inline scripts/linting/Validate-SkillStructure.ps1:173]: The validation confirms `fuzz_harness.py` exists but doesn't verify it actually contains `import atheris`. A skill could pass validation with an empty `fuzz_harness.py`, defeating the Scorecard purpose. The existing `pyproject.toml` checks validate content via regex — this check only tests file existence.

File: scripts/linting/Validate-SkillStructure.ps1 [hunk 8]
```diff
@@ -168,6 +168,14 @@ function Test-PythonSkillConfig {
...
+        if (-not (Test-Path $fuzzHarnessPath -PathType Leaf)) {
+            $errors.Add("$RelativePath - missing tests/fuzz_harness.py (fuzz harness convention for Scorecard compliance)")
+        }
```
Comment (auyidi1) [inline scripts/linting/Validate-SkillStructure.ps1:176]: The validator enforces `fuzz_harness.py` existence but doesn't verify the companion `pyproject.toml` changes (the fuzz dependency group and `python_files` config). The PR description and `copilot-instructions.md` state all three are required together. A skill could have the harness file but miss the pytest discovery config, meaning `uv run pytest` wouldn't actually collect the fuzz tests.

File: scripts/linting/Validate-SkillStructure.ps1 (hunk 7 above)
Comment (WilliamBerryiii) [reply scripts/linting/Validate-SkillStructure.ps1:173]: Agreed. Added content-level validation — `Test-PythonSkillConfig` now checks that `pyproject.toml` includes a `fuzz` dependency group and `fuzz_harness.py` in `python_files` when the harness file exists. This closes the gap between file existence and actual Scorecard compliance.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py (hunk 1 above)
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py:23]: Good catch — the bare `instrument_imports()` call is indeed a no-op here (discarded context manager + modules already loaded).

Replaced with `atheris.instrument_all()` which uses `gc.get_objects()` to instrument all already-loaded Python functions in-place. This is simpler than the dual-import-block approach and the Atheris docs recommend placing it right before `atheris.Setup()`, which is exactly what we have here.

File: scripts/linting/Validate-SkillStructure.ps1 (hunk 8 above)
Comment (WilliamBerryiii) [reply scripts/linting/Validate-SkillStructure.ps1:176]: Fixed — the two companion checks (fuzz dep group regex and `python_files` regex) are now added in the `else` branch after confirming `fuzz_harness.py` exists. This ensures all three documented MUST requirements are enforced together.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+    except (ValueError, IndexError):
```
[the remark this answers is not in the document]
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py]: Intentional fuzz harness pattern. The `except (ValueError, IndexError): pass` blocks suppress expected parse/conversion failures from fuzzed input so the fuzzer continues probing for *unexpected* crashes. Any exception type not in that tuple surfaces as a real finding. This is standard Atheris practice — the narrow catch ensures the harness distinguishes expected failures from genuine bugs.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+        resolve_color(theme_ref)
+    except (ValueError, IndexError):
+        pass
+    theme_dict = {
+        "theme": fdp.ConsumeUnicodeNoSurrogates(15),
+        "brightness": fdp.ConsumeFloatInRange(-1.0, 1.0),
+    }
+    try:
+        resolve_color(theme_dict)
+    except (ValueError, IndexError):
```
[the remark this answers is not in the document]
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py]: Same as above — intentional fuzz harness pattern. The `except (ValueError, IndexError): pass` suppresses expected parse failures from fuzzed hex strings so the fuzzer continues probing for unexpected crashes.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+        pass
+    nested_dict = {"color": "#" + fdp.ConsumeUnicodeNoSurrogates(6)}
+    try:
+        resolve_color(nested_dict)
+    except (ValueError, IndexError):
```
[the remark this answers is not in the document]
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py]: Same as above — intentional fuzz harness pattern. The `except (ValueError, IndexError): pass` suppresses expected parse failures from fuzzed hex strings so the fuzzer continues probing for unexpected crashes.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+        pass
+
+
+def fuzz_hex_brightness(data):
+    """Fuzz hex_brightness with arbitrary strings."""
+    fdp = atheris.FuzzedDataProvider(data)
+    hex_str = fdp.ConsumeUnicodeNoSurrogates(10)
+    try:
+        hex_brightness(hex_str)
+    except (ValueError, IndexError):
```
[the remark this answers is not in the document]
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py]: Same as above — intentional fuzz harness pattern. The `except (ValueError, IndexError): pass` suppresses expected parse failures from fuzzed hex strings so the fuzzer continues probing for unexpected crashes.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+    fdp = atheris.FuzzedDataProvider(data)
+    hex_str = "#" + fdp.ConsumeUnicodeNoSurrogates(6)
+    try:
+        resolve_color(hex_str)
+    except (ValueError, IndexError):
+        pass
+    theme_ref = "@" + fdp.ConsumeUnicodeNoSurrogates(20)
+    try:
+        resolve_color(theme_ref)
+    except (ValueError, IndexError):
```
[the remark this answers is not in the document]
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py]: Same as above — intentional fuzz harness pattern. The `except (ValueError, IndexError): pass` suppresses expected parse failures from fuzzed hex strings so the fuzzer continues probing for unexpected crashes.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py (hunk 3 above)
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py:83]: Valid enhancement. The current target always constructs a fully-formed dict with both `slides` and `deck_issues` keys, so it wouldn't discover missing-key crashes. Adding variants that omit one or both keys is the right approach. Tracking as a follow-up alongside the other coverage expansions (additional fuzz targets, corpus seeds).

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py (hunk 2 above)
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py:121]: Agreed the `atheris` reference could confuse future contributors looking at the function signatures in isolation. That said, these functions are only reachable through `fuzz_dispatch` → `atheris.Setup()` inside the `if __name__ == "__main__" and FUZZING:` guard. No pytest test calls them, and Python doesn't evaluate function bodies at definition time, so `NameError` can't occur during import or test collection.

The annotation suggestion is a good one — will consider adding a clarifying comment above the fuzz target block in a follow-up.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py (hunk 6 above)
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py:35]: Correct analysis — and the behavior you describe (a new exception type surfacing as a crash) is exactly what we want. The `(ValueError, IndexError)` tuple is intentionally narrow to match the actual exception spectrum from `int(hex_str, 16)` and string slicing. If `resolve_color` starts raising `TypeError` or `KeyError` on fuzzed input, that's a genuine finding the fuzzer should report, not a false positive. The "expected parse failures" comment refers specifically to these two known exception types from hex conversion paths.

File: .github/skills/experimental/powerpoint/tests/fuzz_harness.py
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/tests/fuzz_harness.py:1]: Good suggestion. A corpus seed directory would meaningfully improve early coverage — the short-hex, invalid-hex, and theme-ref seeds you outlined map directly to the interesting branches. Tracking this as a follow-up enhancement since the bare harness is sufficient for OSSF Scorecard compliance (Phase 3 only checks for `import atheris`).

File: docs/security/fuzzing.md (hunk 4 above)
Comment (WilliamBerryiii) [reply docs/security/fuzzing.md]: Good catch on the precision. The current wording could imply overall Scorecard score = 10/10. Happy to accept the suggested rewording — it accurately describes Fuzzing as one weighted sub-check contributing to the aggregate score.

File: .github/skills/experimental/powerpoint/pyproject.toml (hunk 5 above)
Comment (WilliamBerryiii) [reply .github/skills/experimental/powerpoint/pyproject.toml:24]: Fair point — the architectural reason for the separate group isn't visible in `pyproject.toml` alone. Happy to add the inline comment to make the manylinux-only constraint self-evident.
